### ISSUE TITLE
The runner of the oracles and the test suite never lets a rule refuse an option, since its configuration carries no `validate` flag

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#540]
### What minimal example or steps are needed to reproduce the bug?

The runner of the oracles and of the test suite, `scripts/harness/lint.ts`, never lets a rule refuse an option. Every run it makes is therefore made with the options handed to it, valid or not, and the guard that was meant to notice — `usable`, which `isUsable` of `scripts/oracles/runs.ts` and `invalidOptionWarnings` of `vitest.setup.ts` both read — cannot fire.

```js
// probe.ts, run from the checkout
import path from "node:path"
import { execFileSync } from "node:child_process"
import { lint } from "./scripts/harness/lint.ts"

let root = execFileSync(`git`, [`rev-parse`, `--show-toplevel`], { encoding: `utf8` }).trim()
let plugin = path.join(root, `lib`, `index.ts`)

// A rule whose primary is a number, handed a keyword
let out = await lint({ code: `a { color: pink; }\n`, config: { plugins: [plugin], rules: { "@stylistic/max-empty-lines": `abc` } } })

console.log(out.results[0].invalidOptionWarnings.length) // 0 — the linter itself answers 1
```

### What did you expect to happen?

What Stylelint answers for the same configuration: one `invalidOptionWarning`, and the rule running over nothing.

```
$ stylelint --config … a.css
Invalid option "abc" for rule "@stylistic/max-empty-lines"
```

### What actually happened?

The rule ran. `stylelint.utils.validateOptions` opens with

```js
const mustValidate = result.stylelint.config?.validate

if (!mustValidate) return true
```

(`node_modules/stylelint/lib/utils/validateOptions.mjs`), and the `config` the runner assembles carries `fix`, `rules` and sometimes `customSyntax` and nothing else. So `validateOptions` hands back `true` for every option of every rule, no rule ever reports an `invalidOption` warning, and every `if (!validOptions) return` in `lib/rules/` is dead under the runner.

Three rules then run past that guard into a crash, since their primary is a number and they build a regular expression out of it:

```
@stylistic/function-max-empty-lines: "abc"    → SyntaxError: Invalid regular expression: /(?:\r\n){abc11,}/u: Incomplete quantifier
@stylistic/selector-max-empty-lines: "abc"    → the same
@stylistic/value-list-max-empty-lines: "abc"  → the same
```

The rest run silently over a stylesheet with an option they would have refused.

### Notes

Nothing measured today is wrong because of it: `scripts/oracles/options.ts` is exactly in step with the plugin — I put all 316 of its rule-and-option rows to the linter, and it refuses none. The defect is that the guard against the two falling out of step is not there. `options.ts` is written out by hand precisely so that a run over an older commit stays comparable, and its own comment says an option a rule does not take is "`options.ts` falling behind the plugin rather than the plugin being wrong" — which is what `isUsable` is meant to catch and cannot.

`make harness-check` does not see it either: it proves the runner and the linter agree over the runs the oracles make, and those all use options the rules accept, where the two do agree.

Found while writing a test for #502 that asks the plugin which rules take a primary conditioned on lineness. That test goes to the linter instead, naming a file so that `needsTheLinter` of `vitest.setup.ts` answers yes.
